testplan/web_ui/testing/src/AssertionPane/AssertionTypes/basicAssertionUtils.js

```diff
@@ -3,6 +3,448 @@ import {hashCode} from '../../Common/utils';
 
 /** @module basicAssertionUtils */
 
+/**
+ * Content required to render a basic assertion.
+ * @typedef {Object} AssertionContent
+ * @property {object|string|null} preTitle - Content above the assertion title
+ * @property {object|string|null} preContent - Content between the preTitle and
+ *                                             the title
+ * @property {object|string|null} leftTitle - Left side of the title
+ * @property {object|string|null} rightTitle - Right side of the title
+ * @property {object|string|null} leftContent - Left side of the main content
+ * @property {object|string|null} rightContent - Right side of the main content
+ * @property {object|string|null} postTitle - Content immediately below the
+ *                                            title
+ * @property {object|string|null} postContent - Final content of assertion
+ */
+
+/**
+ * Prepare the content for the Log assertion.
+ * @param {object} assertion
+ * @param {AssertionContent} defaultContent
+ * @return {AssertionContent} Content for Log assertion
+ * @private
+ */
+function prepareLogContent(assertion, defaultContent) {
+  const preContent = (
+    <span>
+      {assertion.message !== undefined ? assertion.message : null}
+     </span>
+  );
+
+  return {
```

Comment:
I like this system of adjusting the default content with `{...defaultContent }` :D much cleaner

Reply:
Thanks, I find the object-spread syntax is really useful in React - particularly when you want to update just a small part of the component's state.

Reply:
I also think that it generally fits into the React "design philosophy" to create an entirely new object instead of mutating the original object here.